tests/virt_autotest/virt_autotest_base.pm

```diff
@@ -194,6 +194,11 @@ sub run_test {
     #FOR S390X LPAR
     if (check_var('ARCH', 's390x')) {
         virt_utils::lpar_cmd("$test_cmd");
+        if (get_var("VIRT_PRJ4_GUEST_UPGRADE")) {
```

Comment:
I am afraid this VIRT_PRJ4_GUEST_UPGRADE block is not suitable to be put in base class. Can you move it to guest_upgrade_run.pm::run() ? Seems put it as last line in this function can work.

Reply:
yes, I had though this changed together. As my purpose, will add upload log for this virt_autotest_base::run_test ->  if s390x ->  virt_utils::lpar_cmd + upload_asset() . let virt_autotest_base::run_test much more commonality for s390x. 
Not just add this upload log for s390x as the last line in guest_upgrade_run.pm::run() 
Also, had made sure that current code changed do not any block under virt_autotest_base::run_test
[s390x gi-guest_developing-on-host_developing-kvm](http://10.67.131.5/tests/413)

Reply:
Sorry. I do not quite get what you mean but can not be convinced, because current code still put s390 prj4 specific code in a base class which should have common code that serve general purpose.

Reply:
yep,  got it. 
Refer to the current code changed in the virt_autotest_base class as below:
```
sub run_test {
...
    my $test_cmd = $self->get_script_run();
    #FOR S390X LPAR
    if (check_var('ARCH', 's390x')) {
        virt_utils::lpar_cmd("$test_cmd");
        if (get_var("VIRT_PRJ4_GUEST_UPGRADE")) {
            #upload the s390x_guest_upgrade_test.log for s390x guest upgrade test
            upload_asset("/tmp/s390x_guest_upgrade_test.log", 1, 1);
            script_run "rm -rf /tmp/s390x_guest_upgrade_test.log";
        }
        return;
    }
...
```
yes, this virt_autotest_base class have common code that serve general purpose. but the current code changed just only effect s390x arch. moreover, just only s390x arch with prj4 test will upload required logs.
I do understand your concern here about this base class. but, I think that current code changed just only s390x prj4 test for this base class.

Reply:
@alice-suse totally make sense here now. Agree on code changes as your mentioned.  For now, had make code updated.
Add the latest validation run:
s390x
wip [prj4_guest_upgrade_sles15sp2_on_sles15sp2-kvm](http://10.67.131.5/tests/418)
wip [prj4_guest_upgrade_sles12sp5_on_sles12sp5-kvm](http://10.67.131.5/tests/419)